tools/system_libs.py

```diff
@@ -987,6 +989,10 @@ def retrieve():
           f = urlopen(url)
           data = f.read()
 
+      if sha512hash:
+        actual_hash = hashlib.sha512(data).hexdigest()
+        if actual_hash != sha512hash:
+          raise RuntimeError('Unexpected hash: ' + actual_hash)
```

Comment:
If we expect developers to update their hashes only after seeing this error, it might be a good idea to add some documentation of that to the error message.

Reply:
Added a suggestion to update the hash in the port file.